<!-- This launch file is automatically generated. -->
<launch>
  <arg name="IMAGE_TOPIC" default="/multisense/left/image_rect_color" />
  <arg name="PANORAMA_IMAGE_TOPIC" default="/chest_camera_remote/image_color" />
  <arg name="POINT_CLOUD_TOPIC" default="/stereo_preprocessing/stereo_downsampled_ground_cloud/output" />
  <arg name="LASER_POINT_CLOUD_TOPIC" default="/laser_preprocess/odom_cloud/output" />
  <group ns="communication">
    <node pkg="resized_image_transport" type="image_resizer" name="color_image_resizer">
      <remap from="~input/image" to="$(arg IMAGE_TOPIC)" />
      <remap from="~output/image" to="image_rect_color" />
      <remap from="~output/width_scale" to="image_width_scale" />
      <remap from="~output/height_scale" to="image_height_scale" />
      <rosparam>
        resize_scale_x: 0.4
        resize_scale_y: 0.4
      </rosparam>
    </node>

    <node pkg="resized_image_transport" type="image_resizer" name="panorama_color_image_resizer">
      <remap from="~input/image" to="$(arg PANORAMA_IMAGE_TOPIC)" />
      <remap from="~output/image" to="panorama_image" />
      <remap from="~output/width_scale" to="panorama_width_scale" />
      <remap from="~output/height_scale" to="panorama_height_scale" />
      <rosparam>
        resize_scale_x: 0.3
        resize_scale_y: 0.3
        use_camera_info: false
      </rosparam>
    </node>
    
    <node pkg="jsk_topic_tools" type="relay" name="point_cloud_relay" >
      <remap from="~input" to="$(arg POINT_CLOUD_TOPIC)" />
      <remap from="~output" to="point_cloud" />
    </node>

    <node pkg="jsk_topic_tools" type="relay" name="laser_cloud_relay" >
      <remap from="~input" to="$(arg LASER_POINT_CLOUD_TOPIC)" />
      <remap from="~output" to="laser_cloud" />
    </node>
  </group>
</launch>
